lib/commands/pull.sh

```diff
@@ -46,17 +35,14 @@ symlink_new_files() {
     local castle=$1
     shift
     local repo="$repos/$castle"
-    local git_out
-    git_out=$(cd "$repo" && git diff --name-only --diff-filter=AR "$BEFORE_PULL_TAG" HEAD -- home 2>/dev/null | wc -l 2>&1)
-    local result=$?
-    # Remove the tag before doing anything else
-    (cd "$repo" && git tag -d "$BEFORE_PULL_TAG" &>/dev/null)
-    if [[ $result -ne 0 ]]; then
-      continue  # Ignore errors, this operation is not mission critical
-    fi
     if [[ ! -d $repo/home ]]; then
       continue;
     fi
+    # @{1} refers to the previous reflog entry on the current branch, which
+    # will be right before the pull
+    if ! git_out=$(cd "$repo" && git diff --name-only --diff-filter=AR '@{1}' HEAD -- home 2>/dev/null | wc -l 2>&1); then
```

Comment:
Damnit, 11 years ago I made this commit: 805e74287c9ab31d1c68ef38d5dac92602f191b4  
I wish I added a better explanation for the change, but I distinctly remember that there was a good reason for not using the reflog. I can't for the life of me remember *why* though.

Reply:
I had the same thought initially, and realized the reflog wouldn't work if there weren't any new revisions to pull, and it would point to before the last commit or pull. That should be avoided by checking that HEAD changed after doing the pull.

Reply:
Yes! That was it! I just now realized it was the same thing you described in the issue. Perfect!

Reply:
The test suite does cover this case: if I revert the changes to `bin/homeshick`, then the test "pull a recently-pulled castle again" fails.